src/units/types.cpp

```diff
@@ -1103,9 +1106,12 @@ void unit_type_data::set_config(const game_config_view& cfg)
 		gui2::dialogs::loading_screen::progress();
 	}
 
-	if(auto abil_cfg = cfg.optional_child("abilities")) {
-		for(const auto& [key, child_cfg] : abil_cfg->all_children_range()) {
-			abilities_map_.emplace(child_cfg["id"], config(key, child_cfg));
+	for(const auto& abil_cfg : cfg.child_range("abilities")) {
+		for(const auto& [key, child_cfg] : abil_cfg.get().all_children_range()) {
+			const std::string& id = child_cfg["unique_id"].str(child_cfg["id"]);
+			if(abilities_map_.find(id) == abilities_map_.end()) {
```

Comment:
The else case should definitely warn as well. Especially since there is an unexpected id fallback mixed in.